modules/artifact/src/main/java/org/mule/runtime/module/artifact/internal/classloader/FeatureFlaggingAwareRegionClassLoader.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.artifact.internal.classloader;
+
+import static org.mule.runtime.api.config.MuleRuntimeFeature.DISABLE_EXPLICIT_GC_WHEN_DISPOSING_ARTIFACT;
+
+import org.mule.runtime.api.config.FeatureFlaggingService;
+import org.mule.runtime.core.api.config.FeatureFlaggingRegistry;
+import org.mule.runtime.module.artifact.api.classloader.ClassLoaderLookupPolicy;
+import org.mule.runtime.module.artifact.api.classloader.RegionClassLoader;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptor;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This version of {@link RegionClassLoader} has a reference to the {@link FeatureFlaggingService} and is capable of modify its
+ * behavior based on feature flagging.
+ */
+public class FeatureFlaggingAwareRegionClassLoader extends RegionClassLoader {
+
+  private static final AtomicBoolean regionClassloaderFeaturesSet = new AtomicBoolean(false);
+
+  /**
+   * Configures the feature flags related to {@link RegionClassLoader}.
+   */
+  public static void configureRegionClassLoaderFeatureFlags() {
+    // TODO: Refactor Feature Flagging Service Initialization
+    if (!regionClassloaderFeaturesSet.getAndSet(true)) {
+      FeatureFlaggingRegistry featureFlaggingRegistry = FeatureFlaggingRegistry.getInstance();
+      featureFlaggingRegistry.registerFeatureFlag(DISABLE_EXPLICIT_GC_WHEN_DISPOSING_ARTIFACT,
+                                                  featureContext -> featureContext.getArtifactMinMuleVersion()
+                                                      .filter(muleVersion -> muleVersion
+                                                          .atLeast(DISABLE_EXPLICIT_GC_WHEN_DISPOSING_ARTIFACT
+                                                              .getEnabledByDefaultSince()))
+                                                      .isPresent());
+    }
+  }
```

Comment:
I stole this fragment from a similar snippet [here](https://github.com/mulesoft/mule/blob/255a790e2a565d71113c181b30dce2b3aff25aff/core/src/main/java/org/mule/runtime/core/internal/profiling/AbstractProfilingService.java#L142).

I couldn't find any ticket tracking this TODO, but it would be nice to know what are the plans. I'm not sure I understand why the registration needs to be manually done when all the metadata to do it automatically is available in the enum. Is it because some FF are being registered with a predicate different than the `minMuleVersion`?

Reply:
I'll create one. Yes, I think when ff flag was created, the idea is to have different predicates to be evaluated. So you have to register what predicate will activate the f (which is not always the mule version).

Reply:
I'll create a ticket for the TODO.